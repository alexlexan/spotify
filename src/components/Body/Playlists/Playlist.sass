.playlist
    padding-bottom: 20px
    h1
        padding-bottom: 20px
    p 
        font-weight: bold
        padding-top: 10px
        font-size: 18px

    &__item:hover
        cursor: pointer
        .playlist__active
            opacity: 1  
            transition: 500ms opacity ease

    &__active
        opacity: 0
        transition: 500ms opacity ease
        position: absolute 
        right: 15px
        bottom: 14px     
    &__active:hover
        color: #1db954

