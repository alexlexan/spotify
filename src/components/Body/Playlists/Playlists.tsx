import React from "react";
import "./Playlist.sass";
import { Link } from "react-router-dom";
import { PlaylistsType } from "../../../store/reducer";

type Props = {
  playlists: PlaylistsType | null;
  title: string;
};

const Playlists: React.FC<Props> = ({ playlists, title }) => {
  if (!playlists?.items.length) {
    return null;
  }

  return (
    <div className="playlist">
      <h1>{title}</h1>
      <div className="grid">
        {playlists?.items.map((item, index) => (
          <Link to={`/playlists/${item.id}/${item.type}`} key={index}>
            <div className="grid__item playlist__item">
              <img src={item.images[0]?.url} alt={item.name} />
              <p>{item.name}</p>
            </div>
          </Link>
        ))}
      </div>
    </div>
  );
};

export default React.memo(Playlists);
