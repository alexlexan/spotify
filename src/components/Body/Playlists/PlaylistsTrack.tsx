import React from "react";
import "./Playlist.sass";
import { useDispatch } from "react-redux";
import { playAndSetTrack, setSongs } from "../../../store/actions";
import { PlaylistsTrackType, Track } from "../../../store/reducer";

type Props = {
  playlists: PlaylistsTrackType;
  title: string;
};

const PlaylistsTrack: React.FC<Props> = ({ playlists, title }) => {
  const dispatch = useDispatch();
  const play = (track: Track) => {
    dispatch(playAndSetTrack(track));
    dispatch(setSongs(null));
  };

  if (!playlists?.items.length) {
    return null;
  }

  return (
    <div className="playlist">
      <h1>{title}</h1>
      <div className="grid">
        {playlists?.items.map((item, index) => (
          <div
            className="grid__item playlist__item"
            key={index}
            onClick={() => play(item)}
          >
            <img src={item.album.images[0]?.url} alt={item.name} />
            <p>{item.name}</p>
          </div>
        ))}
      </div>
    </div>
  );
};

export default PlaylistsTrack;
