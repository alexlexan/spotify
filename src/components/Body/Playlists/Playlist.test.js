import React from "react";
import Playlists from "./Playlists";

describe("tests Playlists component", () => {
  it("should render Playlists without props", () => {
    const component = shallow(<Playlists />);
    expect(component).toMatchSnapshot();
  });
  it("should render Playlists with props", () => {
    const firstTrack = {
      id: 1,
      name: "firstTrack",
      type: "Track",
      images: [{ url: "firstTrackUrl" }],
    };
    const secondTrack = {
      id: 1,
      name: "secondTrack",
      type: "Track",
      images: [{ url: "secondTrack" }],
    };
    const playlists = {
      items: [firstTrack, secondTrack],
    };
    const component = shallow(
      <Playlists playlists={playlists} title={"test"} />
    );
    expect(component).toMatchSnapshot();
  });
});
