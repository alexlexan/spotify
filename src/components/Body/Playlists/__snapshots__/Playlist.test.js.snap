// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests Playlists component should render Playlists with props 1`] = `
<div
  className="playlist"
>
  <h1>
    test
  </h1>
  <div
    className="grid"
  >
    <Link
      key="0"
      to="/playlists/1/Track"
    >
      <div
        className="grid__item playlist__item"
      >
        <img
          alt="firstTrack"
          src="firstTrackUrl"
        />
        <p>
          firstTrack
        </p>
      </div>
    </Link>
    <Link
      key="1"
      to="/playlists/1/Track"
    >
      <div
        className="grid__item playlist__item"
      >
        <img
          alt="secondTrack"
          src="secondTrack"
        />
        <p>
          secondTrack
        </p>
      </div>
    </Link>
  </div>
</div>
`;

exports[`tests Playlists component should render Playlists without props 1`] = `""`;
